runtime-manager/src/runtime_manager_icecap.rs

```diff
@@ -52,75 +53,83 @@ fn main(config: Config) -> Fallible<()> {
     debug_println!("icecap-realmos: initializing...");
 
     // enable ring buffer to serial-server
-    let virtio_console_client =
-        RingBuffer::unmanaged_from_config(&config.virtio_console_server_ring_buffer);
-    virtio_console_client.enable_notify_read();
-    virtio_console_client.enable_notify_write();
+    let virtio_console_client = RingBuffer::unmanaged_from_config(
+        &config.virtio_console_server_ring_buffer
+    );
     debug_println!("icecap-realmos: enabled ring buffer");
 
     debug_println!("icecap-realmos: running...");
     RuntimeManager::new(
         virtio_console_client,
         config.event_nfn,
-        config.badges.virtio_console_server_ring_buffer,
+        config.badges.virtio_console_server_tx,
+        config.badges.virtio_console_server_rx,
     )
     .run()
 }
 
 struct RuntimeManager {
-    channel: RingBuffer,
+    rb: BufferedRingBuffer,
     event: Notification,
-    virtio_console_server_ring_buffer_badge: Badge,
+    #[allow(dead_code)]
+    virtio_console_server_tx_badge: Badge,
+    virtio_console_server_rx_badge: Badge,
     active: bool,
 }
 
 impl RuntimeManager {
     fn new(
-        channel: RingBuffer,
+        rb: RingBuffer,
         event: Notification,
-        virtio_console_server_ring_buffer_badge: Badge,
+        virtio_console_server_tx_badge: Badge,
+        virtio_console_server_rx_badge: Badge,
     ) -> Self {
         Self {
-            channel: channel,
+            rb: BufferedRingBuffer::new(rb),
```

Comment:
Is there a reason why `new` takes a `RingBuffer` instead of a `BufferedRingBuffer`?

Reply:
`RingBuffer` is the underlying IceCap API we need to communicate over. `BufferedRingBuffer` is really a utility class that `RuntimeManager` uses to help manage the traffic on the ring-buffer.

Moving the `BufferedRingBuffer` out would just add an extra piece of work to the user of the API, and keeping it inside the `RuntimeManager` details allows it to change in the future without an API change.

At some point we may want to consider changing this to take a `Read + Write` type or something more generic, but at the moment it's not clear to me what that would be. `Read` and `Write` require std, and there is a `no_std` library for it, but if we want to use that or an IceCap specific trait is an open question.

Reply:
Ok great, thanks for clarifying.  I'll mark this as resolved.